tests/plotting/test_widgets.py

```diff
@@ -9,17 +9,20 @@
     pytestmark = pytest.mark.skip
 
 
+def noop_callback(*args, **kwargs):
+    """A callback that doesn't do anything."""
+    pass
```

Comment:
This is the most opinionated choice. Most of the callbacks in this module were fine, but 2 of them weren't. I decided that it's cleaner to use a catch-all callback instead of defining a one-arg, a two-arg etc. separate callbacks. Part of the reason for this was that the existing "OK" callbacks return non-`None` things, which aren't so great callbacks after all.

If we want to keep testing the signature of callbacks I can separate these. But the fact that these warnings have been around means getting the signature wrong is not a very strong signal anyway.

Reply:
On second thought, we have cases such as `p.add_spline_widget(callback=noop_callback, pass_widget=True, color=None, show_ribbon=True)` in `test_widgets.py`, and the expected behaviour is exactly that the signature of the callback has to change... so perhaps I should revert (correct) signature-specific callbacks.

Reply:
Are these callback functions called at least once when initialing the widget?  If so, we could use mock objects in the test and then assert that they are called with the expected signature for that test.

Reply:
I'm not sure it's on initialization, but at least some of the callbacks get called some of the time, hence the warning when the signatures are off. I'll look into this, thanks.